### PR TITLE
Refactor Resource volume attach and documentation

### PR DESCRIPTION
**Pull request by Diptipowervs**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccXXX'

...
```

### CODE REVIEW

File: ibm/service/power/ibm_pi_constants.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ const (
 	Arg_CloudInstanceID                     = "pi_cloud_instance_id"
 	Arg_ImageName                           = "pi_image_name"
 	Arg_InstanceName                        = "pi_instance_name"
+	Arg_InstanceId                          = "pi_instace_id"
```
Comment (michaelkad) [inline ibm/service/power/ibm_pi_constants.go:11]: Already have  `Arg_PVMInstanceId` define

File: ibm/service/power/resource_ibm_pi_volume_attach.go [hunk 2]
```diff
@@ -34,29 +33,29 @@ func ResourceIBMPIVolumeAttach() *schema.Resource {
...
+				ForceNew:    true,
+				Required:    true,
+				Type:        schema.TypeString,
 			},
 
-			helpers.PIInstanceId: {
-				Type:        schema.TypeString,
-				Required:    true,
-				ForceNew:    true,
+			Arg_InstanceId: {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_volume_attach.go:50]: Use `Arg_PVMInstanceId`

File: ibm/service/power/resource_ibm_pi_volume_attach.go [hunk 3]
```diff
@@ -34,29 +33,29 @@ func ResourceIBMPIVolumeAttach() *schema.Resource {
...
 				Description: "PI Instance Id",
+				ForceNew:    true,
+				Required:    true,
+				Type:        schema.TypeString,
 			},
 
 			// Computed Attribute
-			helpers.PIVolumeAttachStatus: {
+			Attr_VolumeAttachStatus: {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_volume_attach.go:58]: This should `Attr_Status = status`

File: ibm/service/power/resource_ibm_pi_volume_attach.go [hunk 4]
```diff
@@ -89,7 +88,7 @@ func resourceIBMPIVolumeAttachCreate(ctx context.Context, d *schema.ResourceData
 		log.Printf("Volume State /Status is  permitted and hence attaching the volume to the instance")
 	}
 
-	if volinfo.State == helpers.PIVolumeAllowableAttachStatus && !*volinfo.Shareable {
+	if volinfo.State == Attr_VolumeAllowableAttachStatus && !*volinfo.Shareable {
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_volume_attach.go:91]: The value of `PIVolumeAllowableAttachStatus     = "in-use"` need to be updated

File: ibm/service/power/resource_ibm_pi_volume_attach.go [hunk 5]
```diff
@@ -121,14 +120,14 @@ func resourceIBMPIVolumeAttachRead(ctx context.Context, d *schema.ResourceData,
...
-	client := st.NewIBMPIVolumeClient(ctx, sess, cloudInstanceID)
+	client := instance.NewIBMPIVolumeClient(ctx, sess, cloudInstanceID)
 
 	vol, err := client.CheckVolumeAttach(pvmInstanceID, volumeID)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	d.Set(helpers.PIVolumeAttachStatus, vol.State)
+	d.Set(Attr_VolumeAttachStatus, vol.State)
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_volume_attach.go:130]: See above comment

File: ibm/service/power/resource_ibm_pi_volume_attach.go [hunk 6]
```diff
@@ -170,12 +169,12 @@ func resourceIBMPIVolumeAttachDelete(ctx context.Context, d *schema.ResourceData
...
 
-func isWaitForIBMPIVolumeAttachAvailable(ctx context.Context, client *st.IBMPIVolumeClient, id, cloudInstanceID, pvmInstanceID string, timeout time.Duration) (interface{}, error) {
+func isWaitForIBMPIVolumeAttachAvailable(ctx context.Context, client *instance.IBMPIVolumeClient, id, cloudInstanceID, pvmInstanceID string, timeout time.Duration) (interface{}, error) {
 	log.Printf("Waiting for Volume (%s) to be available for attachment", id)
 
 	stateConf := &resource.StateChangeConf{
-		Pending:    []string{"retry", helpers.PIVolumeProvisioning},
-		Target:     []string{helpers.PIVolumeAllowableAttachStatus},
+		Pending:    []string{"retry", Attr_VolumeProvisioning},
+		Target:     []string{Attr_VolumeAllowableAttachStatus},
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_volume_attach.go:177]: Need to update the const to be those
```
	PIVolumeProvisioning     = "creating"
	PIVolumeAllowableAttachStatus     = "in-use"
```

**Review by michaelkad — CHANGES_REQUESTED** [review]

File: ibm/service/power/resource_ibm_pi_volume_attach.go (hunk 3 above)
Comment (Diptipowervs) [reply ibm/service/power/resource_ibm_pi_volume_attach.go]: Added

**Comment by michaelkad** [comment]
Opened #133

**Pull request closed without merging**